azure-iot-device/azure/iot/device/common/http_transport.py

```diff
@@ -0,0 +1,110 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+import uuid
+import threading
+import json
+import ssl
+from . import transport_exceptions as exceptions
+from .pipeline import pipeline_thread
+from six.moves import http_client
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPTransport(object):
+    """
+    A wrapper class that provides an implementation-agnostic HTTP interface.
+    """
+
+    def __init__(self, hostname, ca_cert=None, x509_cert=None):
+        """
+        Constructor to instantiate an HTTP protocol wrapper.
+
+        :param str hostname: Hostname or IP address of the remote host.
+        :param str ca_cert: Certificate which can be used to validate a server-side TLS connection (optional).
+        :param x509_cert: Certificate which can be used to authenticate connection to a server in lieu of a password (optional).
+        """
+        self._hostname = hostname
+        self._ca_cert = ca_cert
+        self._x509_cert = x509_cert
+        self._ssl_context = self._create_ssl_context()
+
+    def _create_ssl_context(self):
+        """
+        This method creates the SSLContext object used to authenticate the connection.
+        """
+        logger.debug("creating a SSL context")
+        ssl_context = ssl.SSLContext(protocol=ssl.PROTOCOL_TLSv1_2)
+
+        if self._ca_cert:
+            ssl_context.load_verify_locations(cadata=self._ca_cert)
+        else:
+            ssl_context.load_default_certs()
+        ssl_context.verify_mode = ssl.CERT_REQUIRED
+        ssl_context.check_hostname = True
+
+        if self._x509_cert is not None:
+            logger.debug("configuring SSL context with client-side certificate and key")
+            ssl_context.load_cert_chain(
+                self._x509_cert.certificate_file,
+                self._x509_cert.key_file,
+                self._x509_cert.pass_phrase,
+            )
+
+        return ssl_context
+
+    @pipeline_thread.invoke_on_http_thread_nowait
+    # TODO: This star syntax is incompatible with Python 2, change it so that the callback is in front of the optional params.
```

Comment:
># TODO: This star syntax is incompatible with Python 2, change it so that the callback is in front of the optional params. [](start = 4, length = 122)

Get rid of this TODO